Simple Icon File Maker/Simple Icon File Maker/Helpers/BitmapImageExtensions.cs

```diff
@@ -0,0 +1,21 @@
+﻿using Microsoft.UI.Xaml.Media.Imaging;
+using System;
+using System.Threading.Tasks;
+using Windows.Storage.Streams;
+
+namespace Simple_Icon_File_Maker.Helpers;
+
+public static class BitmapImageExtensions
+{
+    public static async Task CheckAndSetSourceAsync(this BitmapImage bitmapImage, MainWindow mainWindow, IRandomAccessStream fileStream)
```

Comment:
I prefer extension methods which take no additional parameters, or only simple parameters like bool, string, char. It seems to be like this functionally would fit better as a private method on `MainWindow.xaml.cs`

something like, but probably not exactly. Like I said in the issue comment `MainWindow.xaml.cs` has some redundant code when it comes to loading new images:

```
private void TryLoadImage(BitmapImage bitmapImage, FileStream fileStream)
{
        try
        {
            await bitmapImage.SetSourceAsync(fileStream);
            await SourceImageUpdated(Path.GetFileName(ImagePath));
        }
        catch (Exception ex)
        {
            ShowErrorInfoBar(ex.Message);
        }
}
```